### DISCUSSION TITLE
How can I get the browser thread when using PROXY_TO_PTHREAD

### DISCUSSION THREAD

**Discussion by allsey87** [emscripten-core/emscripten#21161] [Q&A] [answered]
I would like to run some code on the browser thread when using `PROXY_TO_PTHREAD`. I see there is a function [emscripten_proxy_sync](https://emscripten.org/docs/api_reference/proxying.h.html#c.emscripten_proxy_sync) but it takes an argument `target_thread` of type `pthread_t`. How can I specify the browser thread to this function?

[This example](https://github.com/emscripten-core/emscripten/blob/main/test/pthread/test_pthread_proxying.c#L626) shows using the `pthread_self` function to get the browser/main thread from the main function, however, in the `PROXY_TO_PTHREAD` scenario, the main function is not running in the browser thread...

**Comment by sbc100** [comment, marked as answer]
You can use `emscripten_main_runtime_thread_id`

**Reply by allsey87** [reply]
Thank you!
